### ISSUE TITLE
UX - Tree selections in the JMX tree are not highlighted

### ISSUE THREAD

**Issue by lhein** [hawtio/hawtio-next#72]
When selecting an item from the JMX tree that item is not highlighted as selected so you end up not seeing what you selected in the tree. The Camel Plugin tree has a solution to this already.

**Comment by joshiraez** [comment]
I'll be doing this one together with the search on JMX tree as it should be quite easy and I can do it together with the functionality i'm working on

**Comment by phantomjinx** [comment]
So I spent some time going down a rabbit hole ...

I noticed that clicking on a folder in the JMX TreeView displays a generic Attribute Page and, if a folder, lists the contents of it. One can then click on each row and effectively move the selection in the JMX TreeView to that node. I experimented with updating the tree to highlight selections in blue, in the same way as the CamelTreeView (using the prop `hasSelectableNodes={true}`). However, there is a problem ....

If the parent folder is collapsed, the external selection from the Attribute Page fires, the node is selected but the parent folder is NOT automatically expanded. I spent a good few hours trying to get it to expand using the property `defaultExpanded` but nothing seemed to work. So just to be clear ...

1. In JMXTreeView, click on `jolokia` to display its Attribute Page;
2. On the Attribute Page, click on `Config`, `Discovery` or `ServerHandler`. The Attribute Page will refresh to the contents of that node and selection in the JMXTreeView will reflect the same node;
3. Repeat step 1 then collapse `jolokia` in the JMXTreeView then repeat step 2. Despite the selection working correctly, `jolokia` does NOT re-expand to display the selection.

Looking back at the angular version of the app, the expansion did occur.

If you want to experiment, I can post a branch with my experiments on it. Otherwise, it may be a separate issue that we have to take up with Patternfly.
